components/dashboard/src/teams/JoinTeam.tsx

```diff
@@ -24,7 +24,21 @@ export default function() {
                 if (!inviteId) {
                     throw new Error('This invite URL is incorrect.');
                 }
-                const team = await getGitpodService().server.joinTeam(inviteId);
+
+                let team;
+                try {
+                    team = await getGitpodService().server.joinTeam(inviteId);
+                } catch (error) {
+                    const message: string | undefined = error && typeof error.message === "string" && error.message;
+                    const regExp = /You are already a member of this team. \((.*)\)/;
+                    const match = message && regExp.exec(message);
+                    if (match && match[1]) {
+                        const slug = match[1];
+                        history.push(`/t/${slug}`);
```

Comment:
**question**: This redirects you to the team's workspaces page. Do you think it's more confusing to redirect to the members list instead as described in https://github.com/gitpod-io/gitpod/issues/5925?

Reply:
Dunno. Let's redirect to the Members list instead. I'll ping you for a ✅

Reply:
Done. Please have another look

Reply:
Looking at this now! 👀